Tasks/Common/azure-arm-rest/azure-arm-common.ts

```diff
@@ -15,9 +15,10 @@ export class ApplicationTokenCredentials {
     private secret: string;
     public armUrl: string;
     public authorityUrl: string;
+    public activeDirectoryResourceId: string;
     private token_deferred: Q.Promise<string>;
 
-    constructor(clientId: string, domain: string, secret: string, armUrl: string, authorityUrl: string) {
+    constructor(clientId: string, domain: string, secret: string, armUrl: string, authorityUrl: string, activeDirectoryResourceId: string) {
         if (!Boolean(clientId) || typeof clientId.valueOf() !== 'string') {
```

Comment:
armUrl is no longer used in this class. Consider removing it. or rename it appropriately instead of adding new field